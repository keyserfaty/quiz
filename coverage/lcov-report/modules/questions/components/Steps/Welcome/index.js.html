<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for modules/questions/components/Steps/Welcome/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">modules/questions/components/Steps/Welcome</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { Text } from 'react-native'
&nbsp;
import Button from '../../../../../components/Button'
import Card from '../../../../../components/EmptyCard'
import StepTemplate from '../../StepTemplate'
&nbsp;
import styles from './statics/styles'
&nbsp;
const Welcome = props =&gt; {
  const { onNextStep, status } = props
&nbsp;
  const footer = (
    &lt;Button
      primary
      rerender={status}
      status={status}
      onPress={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >onNextStep(1)}</span>
    &gt;Begin&lt;/Button&gt;
  )
&nbsp;
  return (
    &lt;StepTemplate footer={footer}&gt;
      &lt;Card&gt;
        &lt;Text style={[styles.title]}&gt;Welcome to the Trivia Challenge!&lt;/Text&gt;
        &lt;Text style={styles.text}&gt;You will be presented with 10 true and false questions.&lt;/Text&gt;
        &lt;Text style={[styles.text, { marginTop: 16 }]}&gt;Can you score 100%?&lt;/Text&gt;
      &lt;/Card&gt;
    &lt;/StepTemplate&gt;
  )
}
&nbsp;
export default Welcome</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 06:19:18 GMT-0500 (EST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
