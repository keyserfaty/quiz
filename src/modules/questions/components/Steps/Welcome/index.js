import React from 'react'
import PropTypes from 'prop-types'
import { Text } from 'react-native'

import Button from '../../../../../components/Button'
import Card from '../../../../../components/EmptyCard'
import StepTemplate from '../../StepTemplate'

import styles from './statics/styles'

const Welcome = props => {
  const { onNextStep, status } = props

  const footer = (
    <Button
      primary
      rerender={status}
      status={status}
      disabled={status !== 'success'}
      onPress={() => onNextStep(1)}
    >Begin</Button>
  )

  return (
    <StepTemplate footer={footer}>
      <Card>
        <Text style={[styles.title]}>Welcome to the Trivia Challenge!</Text>
        <Text style={styles.text}>You will be presented with 10 true and false questions.</Text>
        <Text style={[styles.text, { marginTop: 16 }]}>Can you score 100%?</Text>
      </Card>
    </StepTemplate>
  )
}

Welcome.propTypes = {
  onNextStep: PropTypes.func, 
  status: PropTypes.string
}

export default Welcome