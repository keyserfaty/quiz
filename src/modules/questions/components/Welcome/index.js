import React from 'react'
import { View, Text } from 'react-native'

import Button from '../../../../components/Button'

const Welcome = props => {
  const { onNextStep } = props

  return (
    <View>
      <Text>Welcome to the Trivia Challenge!</Text>
      <Text>You will be presented with 10 true and false questions.</Text>
      <Text>Can you score 100%?</Text>
      <Button primary onPress={() => onNextStep(1)}>Begin</Button>
    </View>
  )
}

export default Welcome