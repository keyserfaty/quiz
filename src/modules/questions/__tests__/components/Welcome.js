import React from 'react'
import Welcome from '../../components/Steps/Welcome/index'

import Shallow from 'react-test-renderer/shallow'

const renderer = new Shallow()

test('Should render', () => {
  const component = renderer.render(<Welcome />)
  expect(component).toMatchSnapshot()
})