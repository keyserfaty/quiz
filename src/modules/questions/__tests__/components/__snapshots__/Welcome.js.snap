// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render 1`] = `
<StepTemplate
  footer={
    <ButtonWithSpinnerContainer
      onPress={[Function]}
      primary={true}
    >
      Begin
    </ButtonWithSpinnerContainer>
  }
  header={
    <Component>
      Welcome to the Trivia Challenge!
    </Component>
  }
>
  <Component>
    You will be presented with 10 true and false questions.
  </Component>
  <Component>
    Can you score 100%?
  </Component>
</StepTemplate>
`;
